lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1063,6 +1060,19 @@ def field_to_label(record, field, label)
       record.key?(field) ? { label => record.delete(field).to_s } : {}
     end
 
+    # For every original_label => new_label pair in the label_map, delete the
+    # original_label from the record if it exists, and extract the value to form
+    # a new map with the new_label as the new key.
+    def fields_to_label(record, label_map)
+      extracted_labels = {}
+      return extracted_labels if label_map.nil? || !label_map.is_a?(Hash)
```

Comment:
Interesting. Rubocop recommended `each_with_object` over `inject` for this case, so I used `each_with_object` instead.